purpl3_rms/src/components/scriptLog/scriptLog.js

```diff
@@ -15,100 +16,74 @@ const ScriptLogPage = (props) => {
     const [currentSTDERR, setCurrentSTDERR] = useState("")
     const [consoleType, setCurrentConsoleType] = useState("STDOUT")
 
+    const [refresh , setRefresh] = useState(true)
+    const [error, setError] = useContext(ErrorContext);
+
     useEffect(() => {
 
-        if (pickedLog === 0) {
-            // TODO add logs
+        if (pickedLog === 0 || !script) {
+            console.log("script is null")
         }
         else{
-            
-            axios.post("/api", {
-                body: {
-                    op: "MANAGE_SCRIPT_LOGS",
-                    data:{
-                    funcOP: "GET_FILE",
-                    data: {
-                        Id: script.ID,
-                        Filetype: consoleType,
-                        FP: 0            
-                    }
-                    }
-                }
-                }).then((res) => {
-                    if(consoleType === "STDOUT"){
-                        setCurrentSTDOUT(res.data.entry)
-                    }else {
-                        setCurrentSTDERR(res.data.entry)
-                    }
-                }).catch((res) =>{
-                    console.log(res)
-                })
-            
+            getScriptLogs(script.ID, 0, consoleType).then(res => {
+              if(consoleType === "STDOUT"){
+                setCurrentSTDOUT(res.data.entry)
+              }else {
+                setCurrentSTDERR(res.data.entry)
+              }
+            }).catch(setError);           
             
         }
     },[consoleType,pickedLog,script]);
 
     useEffect(() => {
-        axios.all([
-            axios.post("/api", {
-                body: {
-                  op: "MANAGE_SCRIPTS",
-                  data:{
-                    funcOP: "GET_ALL",
-                    data: {}
-                  }
-                }
-                }),
-            axios.post("/api", {
-                body: {
-                    op: "MANAGE_COMPUTER",
-                    data:{
-                    funcOP: "GET_ALL",
-                    data: {}
+      //only refresh if refresh is true
+      async function retrieveScriptLogs(){
+        let scriptLogs = null;
+        let scripts = null;
+        let computers = null;
+
+        try{
+          scriptLogs = (await getAllScriptLogs()).data.entries;
+          scripts = (await getAllScripts()).data.entries;
+          computers = (await getAllComputers()).data.entries;
+
+        }catch(err) {
+          setError(err);
+          setRefresh(!refresh);
```

Comment:
shouldn't this just be set to false? WHat if its already false and this sets it to true even if an error occurs